In file included from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include/ztest.h:48:0,
                 from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/net/mld/src/main.c:16:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/net/mld/src/main.c: In function ‘test_main’:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/net/mld/src/main.c:452:21: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
     ztest_unit_test(join_group),
                     ^
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include/ztest_test.h:108:3: note: in definition of macro ‘ztest_test_suite’
   __VA_ARGS__, { 0 } \
   ^~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include/ztest_test.h:89:2: note: in expansion of macro ‘ztest_unit_test_setup_teardown’
  ztest_unit_test_setup_teardown(fn, unit_test_noop, unit_test_noop)
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/net/mld/src/main.c:452:5: note: in expansion of macro ‘ztest_unit_test’
     ztest_unit_test(join_group),
     ^~~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/net/mld/src/main.c:452:21: note: (near initialization for ‘_net_mld_test[1].test’)
     ztest_unit_test(join_group),
                     ^
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include/ztest_test.h:108:3: note: in definition of macro ‘ztest_test_suite’
   __VA_ARGS__, { 0 } \
   ^~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include/ztest_test.h:89:2: note: in expansion of macro ‘ztest_unit_test_setup_teardown’
  ztest_unit_test_setup_teardown(fn, unit_test_noop, unit_test_noop)
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/net/mld/src/main.c:452:5: note: in expansion of macro ‘ztest_unit_test’
     ztest_unit_test(join_group),
     ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/Makefile.build:183: recipe for target 'src/main.o' failed

--- a/tests/net/mld/src/main.c
+++ b/tests/net/mld/src/main.c
@@ -49,6 +49,7 @@ static bool is_join_msg_ok;
 static bool is_leave_msg_ok;
 static bool is_query_received;
 static bool is_report_sent;
+static bool ignore_already;
 static struct k_sem wait_data;
 
 #define WAIT_TIME 500
@@ -200,7 +201,12 @@ static void join_group(void)
 
 	ret = net_ipv6_mld_join(iface, &mcast_addr);
 
-	zassert_equal(ret, 0, "Cannot join IPv6 multicast group");
+	if (ignore_already) {
+		zassert_true(ret == 0 || ret == -EALREADY,
+			     "Cannot join IPv6 multicast group");
+	} else {
+		zassert_equal(ret, 0, "Cannot join IPv6 multicast group");
+	}
 
 	k_yield();
 }
@@ -221,6 +227,8 @@ static void leave_group(void)
 static void catch_join_group(void)
 {
 	is_group_joined = false;
+
+	ignore_already = false;
 
 	join_group();
 
@@ -255,6 +263,8 @@ static void catch_leave_group(void)
 static void verify_join_group(void)
 {
 	is_join_msg_ok = false;
+
+	ignore_already = false;
 
 	join_group();
 
@@ -389,6 +399,8 @@ static void verify_send_report(void)
 
 	is_query_received = false;
 	is_report_sent = false;
+
+	ignore_already = true;
 
 	join_group();
 

